### DISCUSSION TITLE
gatewayのディレクトリ構成について

### DISCUSSION THREAD

**Discussion by hrtk7623** [harusys/newsh#21] [Q&A]
@harusys 
gatewayの方は「init.py」と「main.py」で分けていないのには何か理由がありますか？
今のところweatherの方はLINE起動で別関数（weather）が呼び出しうまくできていなくて，関連するところあるのかなと思って気になっています！
ブラウザからはうまくできているのですが！
✕：「LINE」→「gateway」→「weather」　※gatewayのinitは動くがmain関数が呼べない
〇：「ブラウザ」→「gateway」→「weather」

https://github.com/harusys/newsh/blob/40330ae7b63b1dec9a8b6d10552219ee84e3b0e3/gateway/timer/__init__.py#L34

**Comment by harusys** [comment]
Gateway はタイマートリガしか作っていないため、

- HTTP で通常は叩けない
- Fast API 導入していない（≒main / init で分ける必要がない）　　※HTTP トリガ入れる場合のみ

です！

ブラウザで ○ になってるのも、ブラウザから HTTP で叩けるわけではなく、Azure ポータルからテスト実行はできる、ってことよね？
であれば「Azure ポータル：○」が正しくて、Azure ポータルからならたしかにタイマートリガを強制的に実行することは可能です。

Gateway に HTTP トリガの口を作る場合、
「リッチメニュー選択したら」「特定のメッセージ（例：Twitter トレンドおしえて）を入力したら」など、
まずは HTTP トリガー条件を整理するところからなので、現時点では実装していないです～

**Reply by harusys** [reply]
> Gateway に HTTP トリガの口を作る場合、
> 「リッチメニュー選択したら」「特定のメッセージ（例：Twitter トレンドおしえて）を入力したら」など、
> まずは HTTP トリガー条件を整理するところからなので、現時点では実装していないです～

本件、#22 で機能追加 Issue 作成して対応中です！
